enhancements/sdk-generate-commands.md

```diff
@@ -0,0 +1,353 @@
+
+# Build the bundle image.
+build-bundle:
+	docker build -f bundle.Dockerfile -t $(BUNDLE_IMG) .
+```
+
+A user can either invoke `make bundle` or `make bundle-build` or both; there is
+no precedence requirement.
+
+[opm-bundle-generate]:https://github.com/operator-framework/operator-registry/blob/master/docs/design/operator-bundle.md#generate-bundle-annotations-and-dockerfile
+[package-manifest-doc]:https://github.com/operator-framework/operator-lifecycle-manager#discovery-catalogs-and-automated-upgrades
+[kb-makefile]:https://github.com/kubernetes-sigs/kubebuilder/blob/7bdcf98/pkg/scaffold/internal/templates/makefile.go#L58-L151
+
+## Design Details
+
+### Test Plan
+
+New project layout scenarios:
+* [BDD-style E2E tests][pr-e2e-new-layout] that perform the "happy path" workflow for both commands.
+* Unit tests for underlying generators.
+
+Old project layout scenarios:
+* Subcommand tests in a similar style to what [currently exists][dir-e2e-old-layout].
+* Unit tests for underlying generators.
+
+### Graduation Criteria
+
+#### New project layout examples
+
+Both new project layout examples assume the following setup preceded them:
+
+```console
+$ mkdir -p $HOME/projects/example-inc/memcached-operator
+$ cd $HOME/projects/example-inc/memcached-operator
+$ operator-sdk init --domain example.com --repo github.com/example-inc/memcached-operator
+$ operator-sdk create api --group cache --version v1alpha1 --kind Memcached --resource --controller
+$ export USERNAME=<username>
+```
+
+##### `generate bundle`
+
+```console
+$ make bundle
+controller-gen "crd:trivialVersions=true" rbac:roleName=manager-role webhook paths="./..." output:crd:artifacts:config=config/crd/bases
+kustomize build config/bundle | operator-sdk generate bundle -q --overwrite --version 0.0.1
+Provide DisplayName for your operator:		
+Provide version for your operator:		
+Provide minKubeVersion for CSV:		
+...
+INFO[0003] Building annotations.yaml                    
+INFO[0003] Building Dockerfile                          
+operator-sdk bundle validate config/bundle
+INFO[0000] Found annotations file                        bundle-dir=/home/user/go/src/github.com/example-inc/memcached-operator/config/bundle
+INFO[0000] Could not find optional dependencies file     bundle-dir=/home/user/go/src/github.com/example-inc/memcached-operator/config/bundle
+INFO[0000] All validation tests have completed successfully  bundle-dir=/home/user/go/src/github.com/example-inc/memcached-operator/config/bundle
+$ make bundle-build BUNDLE_IMG=quay.io/$USERNAME/memcached-operator-bundle:v0.0.1
+$ make docker-push IMG=quay.io/$USERNAME/memcached-operator-bundle:v0.0.1
+```
+
+##### `generate packagemanifests`
+
+```console
+$ make manifests
+controller-gen "crd:trivialVersions=true" rbac:roleName=manager-role webhook paths="./..." output:crd:artifacts:config=config/crd/bases
+$ kustomize build config/packages | operator-sdk generate packagemanifests -q --channel beta --version 0.0.1
```

Comment:
Should we run `$ kustomize build config/packages | operator-sdk generate packagemanifests -q --channel beta --version 0.0.1` ? Could we not be a new target `make bundle-manifest`for example?

Reply:
We're going to transition away from package manifests at some point, although we don't have a concrete timeline. Regardless I don't think it's a good idea to add this target by default. There will be docs explaining how to add this to the Makefile at the very least.